hearth/lib/hearth/refreshing_identity_resolver.rb

```diff
@@ -0,0 +1,14 @@
+# frozen_string_literal: true
+
+module Hearth
+  # Refreshing Identity resolver that uses a proc to resolve an Identity.
+  # Automatically refreshes the identity if it is near expiration.
+  class RefreshingIdentityResolver < IdentityResolver
```

Comment:
This seems like it could be something like.... ProcIdentityResolver or GenericIdentityResolver.  (naming is hard)

Do we actually need this?  I assume most IdentityResolvers will include the RefreshingIdentity module, and provide an implementation of refresh themselves (as we do in V3).

Reply:
We have IdentityResolver and RefreshingIdentityResolver. IdentityResolver doesn't care for expiration (calls a proc to return a thing) and RefreshingIdentityResolver has a slightly different interface (implement refresh method) and checks expiration. I think it's valid to have both types.. The non-aws identities (login, bearer, api) are probably things that won't need to refresh and probably should not implement "refresh" method, it's weird.

Reply:
Removing RefreshingIdentityResolver and renaming RefreshingIdentity to RefreshingIdentityResolver